packages/rpc/src/methods/txnstatus.ts

```diff
@@ -0,0 +1,27 @@
+import { requestFromNearRpc } from "../util";
+
+export type TxnStatus = "pending" | "success" | "failure";
+
+export const getTxnStatus = async (
+  txnHash: string,
+  senderId: string
+): Promise<TxnStatus> => {
+  const res = await requestFromNearRpc({
+    jsonrpc: "2.0",
+    id: "dontcare",
+    method: "tx",
+    params: [txnHash, senderId],
+  });
+  if (res.error) throw res.error;
+
+  let failure = false;
+  let pending = false;
+  res.result.receipts_outcome.forEach((outcome: any) => {
```

Comment:
is there a situation where result  is undefined and result.receipts_outcome are not an array?